cloud/disk_manager/internal/pkg/dataplane/replicate_disk_task.go

```diff
@@ -137,6 +139,10 @@ func (t *replicateDiskTask) createProxyOverlayDisk(
 	currentCheckpointID string,
 ) (string, error) {
 
+	if !t.disksConfig.GetUseProxyOverlayDisksInRelocation() {
```

Comment:
Flag should be remembered inside replicate task request/state, see https://github.com/ydb-platform/nbs/blob/main/cloud/disk_manager/internal/pkg/dataplane/protos/create_snapshot_from_disk_task.proto#L20